cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -673,9 +677,9 @@ instance HasTextFormatter (DebugPeerSelection SockAddr) where
   -- format.
   formatText _ obj = pack (show obj)
 
-instance Transformable Text IO (PeerSelectionActionsTrace SockAddr) where
+instance Show lAddr => Transformable Text IO (PeerSelectionActionsTrace SockAddr lAddr) where
   trTransformer = trStructuredText
-instance HasTextFormatter (PeerSelectionActionsTrace SockAddr) where
+instance Show lAddr => HasTextFormatter (PeerSelectionActionsTrace SockAddr lAddr) where
```

Comment:
This is exactly what users are complaining about: https://github.com/input-output-hk/cardano-node/issues/4630
Please do not use `show`.

Reply:
I'm not really sure what to do about this. The implementation was using `show` before but now network added an extra argument so now it requires a constraint too. I think someone from the node team should deal with this in another PR.

Reply:
I think the networking team should provide a suitable instance. This is not the node team's responsibility and if it's not done now it will never be done. We don't have the context to render these things appropriately.

Reply:
We're going to make a separate PR before the node 1.36.0 release where we'll switch to using network on CHaP. I can ask network to fix this before they make the release on CHaP, but we should get this PR merged in the meantime.

Reply:
The type variable should be called `versionNumber` not `lAddr`.

@Jimbo4350 it's larger change for us to provide better log messages, I'd expect that we'll use `Show` still for some time in various place (note that we use `json` in quite a few places already).

Reply:
Anyway thanks for reminding about #4630.  I commented there, where the actual problem comes from (I still think that we should migrate from `Show` anyway).

Reply:
I fixed the naming of the type parameter. I'm marking this as resolved for now.

Reply:
I don't think anyone is dumping it on node, but it's outside the scope of this PR. We're only bumping the dependencies here and making sure everything builds. 

We're not creating a release from this PR directly so there's still a chance to work on details after this gets merged.